server/src/test/java/io/crate/integrationtests/disruption/replication/logical/SubscriptionDisruptionIT.java

```diff
@@ -79,4 +90,132 @@ public void test_subscription_state_on_restore_failure() throws Exception {
             transportServices.forEach(MockTransportService::clearAllRules);
         }
     }
+
+    @Test
+    public void test_subscription_metadata_tracker_retries_on_publisher_disconnect() throws Exception {
+        String subscriptionName = "sub1";
+        executeOnPublisher("CREATE TABLE doc.t1 (id INT) CLUSTERED INTO 1 SHARDS WITH(" + defaultTableSettings() +")");
+        createPublication("pub1", false, List.of("doc.t1"));
+        executeOnPublisher("INSERT INTO doc.t1 (id) VALUES (1), (2)");
+        executeOnSubscriber("CREATE SUBSCRIPTION " + subscriptionName +
+                            " CONNECTION '" + publisherConnectionUrl() + "' publication pub1");
+
+        var replicationService = subscriberCluster.getInstance(LogicalReplicationService.class);
```

Comment:
Minor, instead of reflection we could use some package visible getter for `metadataTracker` and `isActive`.

Reply:
I'm not a big fan of making vars accessible for testing purposes only. Maybe personal preference. If you don't have a strong opinion/reason, I'd leave it like that.